src/modules/landing_target_estimator/LandingTargetEstimator.cpp

```diff
@@ -0,0 +1,313 @@
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/*
+ * @file LandingTargetEstimator.cpp
+ *
+ * @author Nicolas de Palezieux (Sunflower Labs) <ndepal@gmail.com>
+ */
+
+#include <px4_config.h>
+#include <px4_defines.h>
+#include <drivers/drv_hrt.h>
+
+#include "LandingTargetEstimator.h"
+
+#define SEC2USEC 1000000.0f
+
+
+namespace landing_target_estimator
+{
+
+LandingTargetEstimator::LandingTargetEstimator() :
+	_targetPosePub(nullptr),
+	_targetInnovationsPub(nullptr),
+	_paramHandle(),
+	_vehicleLocalPosition_valid(false),
+	_vehicleAttitude_valid(false),
+	_sensorCombined_valid(false),
+	_new_irlockReport(false),
+	_estimator_initialized(false),
+	_faulty(false),
+	_last_predict(0),
+	_last_update(0)
+{
+	_paramHandle.acc_unc = param_find("LTEST_ACC_UNC");
+	_paramHandle.meas_unc = param_find("LTEST_MEAS_UNC");
+	_paramHandle.pos_unc_init = param_find("LTEST_POS_UNC_IN");
+	_paramHandle.vel_unc_init = param_find("LTEST_VEL_UNC_IN");
+	_paramHandle.mode = param_find("LTEST_MODE");
+	_paramHandle.scale_x = param_find("LTEST_SCALE_X");
+	_paramHandle.scale_y = param_find("LTEST_SCALE_Y");
+
+	// Initialize uORB topics.
+	_initialize_topics();
+
+	_check_params(true);
+}
+
+LandingTargetEstimator::~LandingTargetEstimator()
+{
+}
+
+void LandingTargetEstimator::update()
+{
+	_check_params(false);
+
+	_update_topics();
+
+	/* predict */
+	if (_estimator_initialized) {
+		if (hrt_absolute_time() - _last_update > landing_target_estimator_TIMEOUT_US) {
+			PX4_WARN("Timeout");
+			_estimator_initialized = false;
+
+		} else if (_vehicleLocalPosition_valid
+			   && _vehicleLocalPosition.v_xy_valid) {
+			float dt = (hrt_absolute_time() - _last_predict) / SEC2USEC;
+
+			// predict target position with the help of accel data
+			matrix::Vector3f a;
+
+			if (_sensorCombined_valid) {
+				matrix::Quaternion<float> q_att(&_vehicleAttitude.q[0]);
+				_R_att = matrix::Dcm<float>(q_att);
+				a = _sensorCombined.accelerometer_m_s2;
```

Comment:
@ndepal You should subscribe to `sensor_bias` and get the accel value from there. I wrote a custom target estimator module, and this improves prediction performance in our case.

 The value in `sensor_bias` is corrected for in-run bias, while `sensor_combined` is not.

Reply:
That's good to know, thanks @mhkabir . I'll change to that.